### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by Shinigami92**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        run: npm run build --if-present
+
+      - name: Test
+        if: github.ref != 'refs/heads/master'
+        run: npm run test
+
+      - name: Test & Coverage
+        if: github.ref == 'refs/heads/master'
+        run: | 
+          npm run test
```
Comment (jerome-benoit) [inline .github/workflows/ci.yml:39]: Is re-running the test required here? It duplicates the previous step.

Comment (Shinigami92) [reply .github/workflows/ci.yml:39]: This will only run when branch is `master`  
Or is `npm run coverage` triggering the test again? :thinking:

Comment (pioardi) [reply .github/workflows/ci.yml:39]: General idea is good, using the branch reference is more clean and easy way.
It is ok for me, waiting for jerome for further feedbacks but for me is ok

Comment (jerome-benoit) [reply .github/workflows/ci.yml:39]: It was just nitpicking on coverage usage and test. 
It's ok for me.

**Comment by Shinigami92** [comment]
@jerome-benoit Do you want to approve this PR?

@pioardi Do you want to merge this PR as the next one?

**Review by jerome-benoit — APPROVED** [review]

**Comment by Shinigami92** [comment]
@pioardi I'm able to merge it myself if someone has approved the PR
Is this your wanted setup? Or do you need to improve some of your repo settings from your side?

image

**Pull request merged**